tools/link.py

```diff
@@ -2561,6 +2561,14 @@ def modularize():
   shared.get_temp_files().note(final_js)
   save_intermediate('modularized')
 
+  if settings.MINIFY_WHITESPACE:
+    # Minify the whitespace in the modularization code that we just generated
+    # Force MAYBE_CLOSURE_COMPILER to false here if we are minifying whitespace
+    # since we know we have already run our internal closure compiler run, and
+    # external closure users will use --minify=0.
+    settings.MAYBE_CLOSURE_COMPILER = 0
```

Comment:
Perhaps to complement this, add this to the PR?
```diff
diff --git a/tools/building.py b/tools/building.py
index bd486cf40b..2aff6575da 100644
--- a/tools/building.py
+++ b/tools/building.py
@@ -543,6 +543,9 @@ def transpile(filename):
 
 @ToolchainProfiler.profile()
 def closure_compiler(filename, advanced=True, extra_closure_args=None):
+  # If we run closure, we must have been aware that it would be run, which
+  # affects codegen in some places.
+  assert settings.MAYBE_CLOSURE_COMPILER
   user_args = []
   env_args = os.environ.get('EMCC_CLOSURE_ARGS')
   if env_args:
```
That will prevent some future refactoring from running closure later.